src/zkp/pimod.rs

```diff
@@ -26,38 +49,44 @@ static LAMBDA: usize = crate::parameters::SOUNDNESS_PARAMETER;
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub(crate) struct PiModProof {
-    w: BigNumber,
+    random_jacobi_one: BigNumber,
     // (x, a, b, z),
     elements: Vec<PiModProofElements>,
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub(crate) struct PiModProofElements {
-    x: BigNumber,
-    a: usize,
-    b: usize,
-    z: BigNumber,
-    y: BigNumber,
+    /// Fourth root of the constraint linking the `random_jacobi_one` request
+    /// with the challenge (`x_i` in the paper).
+    fourth_root: BigNumber,
+    /// Determining the sign of the root (`a_i` in the paper).
+    sign_exponent: usize,
+    /// Satifies the jacobi symbol (`b_i` in the paper).
+    jacobi_exponent: usize,
+    /// Function of the challenge and the secret input (`z_i` in the paper).
+    challenge_secret_link: BigNumber,
+    /// Challenge value, chosen via fiat-Shamir (`y_i` in the paper).
+    challenge: BigNumber,
 }
 
 #[derive(Serialize)]
-pub(crate) struct PiModInput {
-    N: BigNumber,
+pub(crate) struct CommonInput {
+    modulus: BigNumber,
 }
 
-impl PiModInput {
+impl CommonInput {
     pub(crate) fn new(N: &BigNumber) -> Self {
-        Self { N: N.clone() }
+        Self { modulus: N.clone() }
```

Comment:
todo: rename the parameter as well. Variables names shouldn't include capital letters.